pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -312,14 +318,55 @@ func (o *UpgradeBootOptions) updateBootConfig(versionStreamURL string, versionSt
 		return errors.Wrapf(err, "failed to fetch upgrade tag %s from %s", upgradeVersion, bootConfigURL)
 	}
 
+	// Set up custom merge driver to ensure that specified files always use the local/dev env version in merges/cherry picks with conflicts
+	err = o.configureGitMergeExcludes()
+	if err != nil {
+		return errors.Wrapf(err, "configuring files (%s) to always use existing version when conflicting with upstream", strings.Join(filesExcludedFromCherryPick, ", "))
+	}
+
 	err = o.cherryPickCommits(configCloneDir, currentSha, upgradeSha)
 	if err != nil {
 		return errors.Wrap(err, "failed to cherry pick upgrade commits")
 	}
-	err = o.excludeFiles(currentSha)
+	return nil
+}
+
+func (o *UpgradeBootOptions) configureGitMergeExcludes() error {
+	err := o.Git().Config(o.Dir, "--local", fmt.Sprintf("merge.%s.name", keepDevEnvKey), keepDevEnvDriverName)
+	if err != nil {
+		return errors.Wrapf(err, "configuring name for exclude merge driver in %s", o.Dir)
+	}
+	err = o.Git().Config(o.Dir, "--local", fmt.Sprintf("merge.%s.driver", keepDevEnvKey), "true")
+	if err != nil {
+		return errors.Wrapf(err, "configuring driver for exclude merge driver in %s", o.Dir)
+	}
+
+	// Read the existing .git/info/attributes if it exists
+	gitAttrFile := filepath.Join(o.Dir, ".git", "info", "attributes")
+	var existingGitAttr string
+	gaExists, err := util.FileExists(gitAttrFile)
+	if err != nil {
+		return errors.Wrapf(err, "checking if repo-local git attributes file %s exists", gitAttrFile)
+	}
+	if gaExists {
+		gaBytes, err := ioutil.ReadFile(gitAttrFile)
+		if err != nil {
+			return errors.Wrapf(err, "reading existing repo-local git attributes from %s", gitAttrFile)
+		}
+		existingGitAttr = string(gaBytes)
+	}
+
+	// Write the existing .git/info/attributes content and marking the selected files as using our custom driver
+	gitAttrContent := existingGitAttr + "\n"
+	for _, excludedFile := range filesExcludedFromCherryPick {
+		gitAttrContent += fmt.Sprintf("%s merge=%s\n", excludedFile, keepDevEnvKey)
+	}
+
+	err = ioutil.WriteFile(gitAttrFile, []byte(gitAttrContent), util.DefaultFileWritePermissions)
```

Comment:
Don't you have to unset this as well after the upgrade? You only want the custom merge driver during `upgrade boot` right? Afterwards, the normal git workflow should apply, right? Does your change now not mean that merging is off for regular evolution of the dev repo?